mlir/lib/Dialect/MIOpen/Generator/IsaNameParser.cpp

```diff
@@ -0,0 +1,128 @@
+//===- IsaNameParser.cpp - MLIR to C++ option parsing ---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements isa name string parser
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/MIOpen/Generator/IsaNameParser.h"
+#include "llvm/Support/Error.h"
+
+#include <cstring>
```

Comment:
could you share insights why `<cstring>` is needed here? I think the inclusion of `<string>` in the header should suffice?

Reply:
`strlen()` is the reason why it is here. It has to be there because of the const c-string: `const char gcnArchDelimiter[] = ":";`

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/f53f5226693392fedcee058aa7b7ff9d0ebdfc2c/mlir/lib/Dialect/MIOpen/Generator/IsaNameParser.cpp#L102